File: android/src/main/java/com/azifacemobile/theme/ScrollIndicator.java
package com.azifacemobile.theme;

import android.graphics.Typeface;

import com.azifacemobile.theme.abstracts.ViewStyle;
import com.azifacemobile.utils.Theme;
import com.facebook.react.bridge.ReactApplicationContext;

import org.json.JSONException;
import org.json.JSONObject;

import javax.annotation.Nullable;

public class ScrollIndicator extends ViewStyle {
  private static final String KEY = "scrollIndicator";
  private final Theme theme;
  private final JSONObject target;
  private final Color color;
  private final Font font;

  public ScrollIndicator(ReactApplicationContext context, JSONObject target) {
    super(target, KEY);

    this.theme = new Theme();
    this.target = this.theme.getTarget(target, KEY);
    this.color = new Color();
    this.font = new Font(context);
  }

  private boolean getBoolean(String key) {
    try {
      if (!this.theme.exists(this.target, key)) {
        return true;
      }

      return this.target.getBoolean(key);
    } catch (JSONException e) {
      return true;
    }
  }

  @Override
  public int getCornerRadius() {
    return this.getCornerRadius(-1);
  }

  @Override
  public int getElevation() {
    return this.getElevation(10);
  }

  public int getBackgroundNormalColor() {
    return this.color.getColor(this.target, "backgroundNormalColor", "#026ff4");
  }

  public int getBackgroundHighlightColor() {
    return this.color.getColor(this.target, "backgroundHighlightColor", "#0264dc");
  }

  public int getForegroundNormalColor() {
    return this.color.getColor(this.target, "foregroundNormalColor", "#ffffff");
  }

  public int getForegroundHighlightColor() {
    return this.color.getColor(this.target, "foregroundHighlightColor", "#ffffff");
  }

  public boolean getEnableScrollIndicator() {
    return this.getBoolean("showsScrollIndicator");
  }

  public boolean getEnableScrollIndicatorTextAnimation() {
    return this.getBoolean("showsScrollTextAnimation");
  }

  public boolean getShowScrollIndicatorImage() {
    return this.getBoolean("showsScrollImage");
  }

  @Nullable
  public Typeface getFont() {
    return this.font.getTypography(this.target, "font");
  }
}

File: ios/Theme/ScrollIndicator.swift
import FaceTecSDK

public class ScrollIndicator: ViewStyle {
  private static let KEY: String = "scrollIndicator"
  private let style: Style
  private let target: NSDictionary?
  private let color: Color
  private let font: Font
  private let shadow: Shadow

  init(target: NSDictionary?) {
    self.style = Style()

    self.target = self.style.getTarget(target, key: ScrollIndicator.KEY)
    self.color = Color()
    self.font = Font()
    self.shadow = Shadow(target: self.target)

    super.init(key: ScrollIndicator.KEY)
  }

  override public func getCornerRadius() -> Int32 {
    return super.getCornerRadius(-1)
  }

  public func getBackgroundNormalColor() -> UIColor {
    return self.color.getColor(self.target, key: "backgroundNormalColor", defaultColor: "#026ff4")
  }

  public func getBackgroundHighlightColor() -> UIColor {
    return self.color.getColor(self.target, key: "backgroundHighlightColor", defaultColor: "#0264dc")
  }

  public func getForegroundNormalColor() -> UIColor {
    return self.color.getColor(self.target, key: "foregroundNormalColor", defaultColor: "#ffffff")
  }

  public func getForegroundHighlightColor() -> UIColor {
    return self.color.getColor(self.target, key: "foregroundHighlightColor", defaultColor: "#ffffff")
  }

  public func getEnableScrollIndicator() -> Bool {
    let key = "showsScrollIndicator"
    if !self.style.exists(self.target, key: key) {
      return true
    }

    return self.target?[key] as? Bool ?? true
  }

  public func getEnableScrollIndicatorTextAnimation() -> Bool {
    let key = "showsScrollTextAnimation"
    if !self.style.exists(self.target, key: key) {
      return true
    }

    return self.target?[key] as? Bool ?? true
  }

  public func getShowScrollIndicatorImage() -> Bool {
    let key = "showsScrollImage"
    if !self.style.exists(self.target, key: key) {
      return true
    }

    return self.target?[key] as? Bool ?? true
  }

  public func getFont() -> UIFont {
    return self.font.getTypography(theme: self.target, key: "font")
  }

  public func getShadow() -> FaceTecShadow {
    return self.shadow.getShadow()
  }
}
